{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "import warnings; \n",
    "warnings.simplefilter('ignore')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 0. Install and Import Dependencies"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: pystan in c:\\users\\vtiwari2\\anaconda3\\lib\\site-packages (2.19.1.1)\n",
      "Requirement already satisfied: numpy>=1.7 in c:\\users\\vtiwari2\\anaconda3\\lib\\site-packages (from pystan) (1.20.1)\n",
      "Requirement already satisfied: Cython!=0.25.1,>=0.22 in c:\\users\\vtiwari2\\anaconda3\\lib\\site-packages (from pystan) (0.29.23)\n",
      "Collecting fbprophet\n",
      "  Using cached fbprophet-0.7.1.tar.gz (64 kB)\n",
      "Requirement already satisfied: Cython>=0.22 in c:\\users\\vtiwari2\\anaconda3\\lib\\site-packages (from fbprophet) (0.29.23)\n",
      "Requirement already satisfied: cmdstanpy==0.9.5 in c:\\users\\vtiwari2\\anaconda3\\lib\\site-packages (from fbprophet) (0.9.5)\n",
      "Requirement already satisfied: pystan>=2.14 in c:\\users\\vtiwari2\\anaconda3\\lib\\site-packages (from fbprophet) (2.19.1.1)\n",
      "Requirement already satisfied: numpy>=1.15.4 in c:\\users\\vtiwari2\\anaconda3\\lib\\site-packages (from fbprophet) (1.20.1)\n",
      "Requirement already satisfied: pandas>=1.0.4 in c:\\users\\vtiwari2\\anaconda3\\lib\\site-packages (from fbprophet) (1.2.4)\n",
      "Requirement already satisfied: matplotlib>=2.0.0 in c:\\users\\vtiwari2\\anaconda3\\lib\\site-packages (from fbprophet) (3.3.4)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "  ERROR: Command errored out with exit status 1:\n",
      "   command: 'C:\\Users\\vtiwari2\\Anaconda3\\python.exe' -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\vtiwari2\\\\AppData\\\\Local\\\\Temp\\\\1\\\\pip-install-7ei2nuta\\\\fbprophet_1b01894b54ea43c0bb0225d22ec02073\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\vtiwari2\\\\AppData\\\\Local\\\\Temp\\\\1\\\\pip-install-7ei2nuta\\\\fbprophet_1b01894b54ea43c0bb0225d22ec02073\\\\setup.py'\"'\"';f=getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__);code=f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' bdist_wheel -d 'C:\\Users\\vtiwari2\\AppData\\Local\\Temp\\1\\pip-wheel-rao3_xkv'\n",
      "       cwd: C:\\Users\\vtiwari2\\AppData\\Local\\Temp\\1\\pip-install-7ei2nuta\\fbprophet_1b01894b54ea43c0bb0225d22ec02073\\\n",
      "  Complete output (285 lines):\n",
      "  running bdist_wheel\n",
      "  running build\n",
      "  running build_py\n",
      "  creating build\n",
      "  creating build\\lib\n",
      "  creating build\\lib\\fbprophet\n",
      "  creating build\\lib\\fbprophet\\stan_model\n",
      "  Importing plotly failed. Interactive plots will not work.\n",
      "  INFO:pystan:COMPILING THE C++ CODE FOR MODEL anon_model_f5236004a3fd5b8429270d00efcc0cf9 NOW.\n",
      "  WARNING:pystan:MSVC compiler is not supported\n",
      "  stanfit4anon_model_f5236004a3fd5b8429270d00efcc0cf9_5748532310016087196.cpp\n",
      "  C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\numpy\\core\\include\\numpy\\npy_1_7_deprecated_api.h(14) : Warning Msg: Using deprecated NumPy API, disable it with #define NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION\n",
      "  C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/prim/mat/meta/seq_view.hpp(145): warning C4267: 'return': conversion from 'size_t' to 'int', possible loss of data\n",
      "  C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/prim/mat/fun/multiply_lower_tri_self_transpose.hpp(19): warning C4244: 'initializing': conversion from 'Eigen::EigenBase<Derived>::Index' to 'int', possible loss of data\n",
      "          with\n",
      "          [\n",
      "              Derived=Eigen::Matrix<double,-1,-1,0,-1,-1>\n",
      "          ]\n",
      "  C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/prim/mat/fun/multiply_lower_tri_self_transpose.hpp(27): warning C4244: 'initializing': conversion from 'Eigen::EigenBase<Derived>::Index' to 'int', possible loss of data\n",
      "          with\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: LunarCalendar>=0.0.9 in c:\\users\\vtiwari2\\anaconda3\\lib\\site-packages (from fbprophet) (0.0.9)\n",
      "Requirement already satisfied: convertdate>=2.1.2 in c:\\users\\vtiwari2\\anaconda3\\lib\\site-packages (from fbprophet) (2.3.2)\n",
      "Requirement already satisfied: holidays>=0.10.2 in c:\\users\\vtiwari2\\anaconda3\\lib\\site-packages (from fbprophet) (0.11.3.1)\n",
      "Requirement already satisfied: setuptools-git>=1.2 in c:\\users\\vtiwari2\\anaconda3\\lib\\site-packages (from fbprophet) (1.2)\n",
      "Requirement already satisfied: python-dateutil>=2.8.0 in c:\\users\\vtiwari2\\anaconda3\\lib\\site-packages (from fbprophet) (2.8.1)\n",
      "Requirement already satisfied: tqdm>=4.36.1 in c:\\users\\vtiwari2\\anaconda3\\lib\\site-packages (from fbprophet) (4.59.0)\n",
      "Requirement already satisfied: pytz>=2014.10 in c:\\users\\vtiwari2\\anaconda3\\lib\\site-packages (from convertdate>=2.1.2->fbprophet) (2021.1)\n",
      "Requirement already satisfied: pymeeus<=1,>=0.3.13 in c:\\users\\vtiwari2\\anaconda3\\lib\\site-packages (from convertdate>=2.1.2->fbprophet) (0.5.11)\n",
      "Requirement already satisfied: korean-lunar-calendar in c:\\users\\vtiwari2\\anaconda3\\lib\\site-packages (from holidays>=0.10.2->fbprophet) (0.2.1)\n",
      "Requirement already satisfied: hijri-converter in c:\\users\\vtiwari2\\anaconda3\\lib\\site-packages (from holidays>=0.10.2->fbprophet) (2.2.2)\n",
      "Requirement already satisfied: ephem>=3.7.5.3 in c:\\users\\vtiwari2\\anaconda3\\lib\\site-packages (from LunarCalendar>=0.0.9->fbprophet) (4.1)\n",
      "Requirement already satisfied: pillow>=6.2.0 in c:\\users\\vtiwari2\\anaconda3\\lib\\site-packages (from matplotlib>=2.0.0->fbprophet) (8.2.0)\n",
      "Requirement already satisfied: pyparsing!=2.0.4,!=2.1.2,!=2.1.6,>=2.0.3 in c:\\users\\vtiwari2\\anaconda3\\lib\\site-packages (from matplotlib>=2.0.0->fbprophet) (2.4.7)\n",
      "Requirement already satisfied: cycler>=0.10 in c:\\users\\vtiwari2\\anaconda3\\lib\\site-packages (from matplotlib>=2.0.0->fbprophet) (0.10.0)\n",
      "Requirement already satisfied: kiwisolver>=1.0.1 in c:\\users\\vtiwari2\\anaconda3\\lib\\site-packages (from matplotlib>=2.0.0->fbprophet) (1.3.1)\n",
      "Requirement already satisfied: six in c:\\users\\vtiwari2\\anaconda3\\lib\\site-packages (from cycler>=0.10->matplotlib>=2.0.0->fbprophet) (1.15.0)\n",
      "Building wheels for collected packages: fbprophet\n",
      "  Building wheel for fbprophet (setup.py): started\n",
      "  Building wheel for fbprophet (setup.py): finished with status 'error'\n",
      "  Running setup.py clean for fbprophet\n",
      "Failed to build fbprophet\n",
      "Installing collected packages: fbprophet\n",
      "    Running setup.py install for fbprophet: started\n",
      "    Running setup.py install for fbprophet: finished with status 'error'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "          [\n",
      "              Derived=Eigen::Matrix<double,-1,-1,0,-1,-1>\n",
      "          ]\n",
      "  C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/prim/mat/fun/csr_u_to_z.hpp(28): warning C4267: 'argument': conversion from 'size_t' to 'int', possible loss of data\n",
      "  C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/prim/mat/fun/matrix_exp_action_handler.hpp(118): warning C4244: 'initializing': conversion from 'double' to 'int', possible loss of data\n",
      "  C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/prim/mat/fun/to_matrix.hpp(136): warning C4267: 'initializing': conversion from 'size_t' to 'int', possible loss of data\n",
      "  C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/prim/mat/fun/to_row_vector.hpp(31): warning C4267: 'initializing': conversion from 'size_t' to 'int', possible loss of data\n",
      "  C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/prim/mat/fun/to_vector.hpp(31): warning C4267: 'initializing': conversion from 'size_t' to 'int', possible loss of data\n",
      "  C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/prim/mat/fun/welford_covar_estimator.hpp(31): warning C4244: 'return': conversion from 'double' to 'int', possible loss of data\n",
      "  C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/prim/mat/fun/welford_var_estimator.hpp(31): warning C4244: 'return': conversion from 'double' to 'int', possible loss of data\n",
      "  C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\src\\stan/io/dump.hpp(243): warning C4101: 'exc': unreferenced local variable\n",
      "  C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\src\\stan/io/dump.hpp(270): warning C4101: 'exc': unreferenced local variable\n",
      "  C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\src\\stan/io/dump.hpp(284): warning C4101: 'exc': unreferenced local variable\n",
      "  C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/meta/operands_and_partials.hpp(42): warning C4267: 'return': conversion from 'size_t' to 'int', possible loss of data\n",
      "  C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/meta/operands_and_partials.hpp(155): warning C4267: 'return': conversion from 'size_t' to 'int', possible loss of data\n",
      "  C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/scal/fun/bessel_first_kind.hpp(19): warning C4244: 'argument': conversion from 'double' to 'int', possible loss of data\n",
      "  C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/scal/fun/bessel_first_kind.hpp(20): warning C4244: 'argument': conversion from 'double' to 'int', possible loss of data\n",
      "  C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/scal/fun/bessel_second_kind.hpp(19): warning C4244: 'argument': conversion from 'double' to 'int', possible loss of data\n",
      "  C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/scal/fun/bessel_second_kind.hpp(20): warning C4244: 'argument': conversion from 'double' to 'int', possible loss of data\n",
      "  C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/scal/fun/modified_bessel_first_kind.hpp(19): warning C4244: 'argument': conversion from 'double' to 'int', possible loss of data\n",
      "  C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/scal/fun/modified_bessel_first_kind.hpp(20): warning C4244: 'argument': conversion from 'double' to 'int', possible loss of data\n",
      "  C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/scal/fun/modified_bessel_second_kind.hpp(19): warning C4244: 'argument': conversion from 'double' to 'int', possible loss of data\n",
      "  C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/scal/fun/modified_bessel_second_kind.hpp(20): warning C4244: 'argument': conversion from 'double' to 'int', possible loss of data\n",
      "  C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/scal/fun/rising_factorial.hpp(18): warning C4244: 'argument': conversion from 'double' to 'int', possible loss of data\n",
      "  C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/fun/cholesky_decompose.hpp(82): warning C4244: 'initializing': conversion from 'Eigen::EigenBase<Derived>::Index' to 'int', possible loss of data\n",
      "          with\n",
      "          [\n",
      "              Derived=Eigen::Matrix<stan::math::var,-1,-1,0,-1,-1>\n",
      "          ]\n",
      "  C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/fun/cholesky_decompose.hpp(194): warning C4244: 'initializing': conversion from 'Eigen::EigenBase<Derived>::Index' to 'int', possible loss of data\n",
      "          with\n",
      "          [\n",
      "              Derived=Eigen::Matrix<stan::math::var,-1,-1,0,-1,-1>\n",
      "          ]\n",
      "  C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/fun/log_softmax.hpp(104): warning C4244: 'argument': conversion from 'Eigen::EigenBase<Derived>::Index' to 'int', possible loss of data\n",
      "          with\n",
      "          [\n",
      "              Derived=Eigen::Matrix<stan::math::var,-1,1,0,-1,1>\n",
      "          ]\n",
      "  C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/fun/multiply_lower_tri_self_transpose.hpp(21): warning C4244: 'initializing': conversion from 'Eigen::EigenBase<Derived>::Index' to 'int', possible loss of data\n",
      "          with\n",
      "          [\n",
      "              Derived=Eigen::Matrix<stan::math::var,-1,-1,0,-1,-1>\n",
      "          ]\n",
      "  C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/fun/multiply_lower_tri_self_transpose.hpp(22): warning C4244: 'initializing': conversion from 'Eigen::EigenBase<Derived>::Index' to 'int', possible loss of data\n",
      "          with\n",
      "          [\n",
      "              Derived=Eigen::Matrix<stan::math::var,-1,-1,0,-1,-1>\n",
      "          ]\n",
      "  C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(410): error C2660: 'stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>::count_memory': function does not take 2 arguments\n",
      "          with\n",
      "          [\n",
      "              F=stan::math::internal::ordered_constrain_op\n",
      "          ]\n",
      "  C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(246): note: see declaration of 'stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>::count_memory'\n",
      "          with\n",
      "          [\n",
      "              F=stan::math::internal::ordered_constrain_op\n",
      "          ]\n",
      "  C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(408): note: while compiling class template member function 'auto stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>::operator ()(const Eigen::Matrix<stan::math::var,-1,1,0,-1,1> &)'\n",
      "          with\n",
      "          [\n",
      "              F=stan::math::internal::ordered_constrain_op\n",
      "          ]\n",
      "  C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(617): note: see reference to class template instantiation 'stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>' being compiled\n",
      "          with\n",
      "          [\n",
      "              F=stan::math::internal::ordered_constrain_op\n",
      "          ]\n",
      "  C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/fun/ordered_constrain.hpp(88): note: see reference to function template instantiation 'auto stan::math::adj_jac_apply<stan::math::internal::ordered_constrain_op,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>(const Eigen::Matrix<stan::math::var,-1,1,0,-1,1> &)' being compiled\n",
      "  C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(412): error C2660: 'stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>::prepare_x_vis': function does not take 1 arguments\n",
      "          with\n",
      "          [\n",
      "              F=stan::math::internal::ordered_constrain_op\n",
      "          ]\n",
      "  C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(389): note: see declaration of 'stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>::prepare_x_vis'\n",
      "          with\n",
      "          [\n",
      "              F=stan::math::internal::ordered_constrain_op\n",
      "          ]\n",
      "  C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(414): error C2664: 'stan::math::var stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>::build_return_varis_and_vars(const double &)': cannot convert argument 1 from 'Eigen::VectorXd' to 'const double &'\n",
      "          with\n",
      "          [\n",
      "              F=stan::math::internal::ordered_constrain_op\n",
      "          ]\n",
      "  C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(414): note: Reason: cannot convert from 'Eigen::VectorXd' to 'const double'\n",
      "  C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(414): note: No user-defined-conversion operator available that can perform this conversion, or the operator cannot be called\n",
      "  C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(333): note: see declaration of 'stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>::build_return_varis_and_vars'\n",
      "          with\n",
      "          [\n",
      "              F=stan::math::internal::ordered_constrain_op\n",
      "          ]\n",
      "  C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/fun/ordered_constrain.hpp(88): error C2440: 'return': cannot convert from 'void' to 'Eigen::Matrix<stan::math::var,-1,1,0,-1,1>'\n",
      "  C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/fun/ordered_constrain.hpp(88): note: Expressions of type void cannot be converted to other types\n",
      "  C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(410): error C2660: 'stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>::count_memory': function does not take 2 arguments\n",
      "          with\n",
      "          [\n",
      "              F=stan::math::internal::positive_ordered_constrain_op\n",
      "          ]\n",
      "  C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(246): note: see declaration of 'stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>::count_memory'\n",
      "          with\n",
      "          [\n",
      "              F=stan::math::internal::positive_ordered_constrain_op\n",
      "          ]\n",
      "  C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(408): note: while compiling class template member function 'auto stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>::operator ()(const Eigen::Matrix<stan::math::var,-1,1,0,-1,1> &)'\n",
      "          with\n",
      "          [\n",
      "              F=stan::math::internal::positive_ordered_constrain_op\n",
      "          ]\n",
      "  C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(617): note: see reference to class template instantiation 'stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>' being compiled\n",
      "          with\n",
      "          [\n",
      "              F=stan::math::internal::positive_ordered_constrain_op\n",
      "          ]\n",
      "  C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/fun/positive_ordered_constrain.hpp(86): note: see reference to function template instantiation 'auto stan::math::adj_jac_apply<stan::math::internal::positive_ordered_constrain_op,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>(const Eigen::Matrix<stan::math::var,-1,1,0,-1,1> &)' being compiled\n",
      "  C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(412): error C2660: 'stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>::prepare_x_vis': function does not take 1 arguments\n",
      "          with\n",
      "          [\n",
      "              F=stan::math::internal::positive_ordered_constrain_op\n",
      "          ]\n",
      "  C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(389): note: see declaration of 'stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>::prepare_x_vis'\n",
      "          with\n",
      "          [\n",
      "              F=stan::math::internal::positive_ordered_constrain_op\n",
      "          ]\n",
      "  C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(414): error C2664: 'stan::math::var stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>::build_return_varis_and_vars(const double &)': cannot convert argument 1 from 'Eigen::VectorXd' to 'const double &'\n",
      "          with\n",
      "          [\n",
      "              F=stan::math::internal::positive_ordered_constrain_op\n",
      "          ]\n",
      "  C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(414): note: Reason: cannot convert from 'Eigen::VectorXd' to 'const double'\n",
      "  C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(414): note: No user-defined-conversion operator available that can perform this conversion, or the operator cannot be called\n",
      "  C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(333): note: see declaration of 'stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>::build_return_varis_and_vars'\n",
      "          with\n",
      "          [\n",
      "              F=stan::math::internal::positive_ordered_constrain_op\n",
      "          ]\n",
      "  C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/fun/positive_ordered_constrain.hpp(86): error C2440: 'return': cannot convert from 'void' to 'Eigen::Matrix<stan::math::var,-1,1,0,-1,1>'\n",
      "  C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/fun/positive_ordered_constrain.hpp(86): note: Expressions of type void cannot be converted to other types\n",
      "  C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(410): error C2660: 'stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>::count_memory': function does not take 2 arguments\n",
      "          with\n",
      "          [\n",
      "              F=stan::math::internal::simplex_constrain_op\n",
      "          ]\n",
      "  C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(246): note: see declaration of 'stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>::count_memory'\n",
      "          with\n",
      "          [\n",
      "              F=stan::math::internal::simplex_constrain_op\n",
      "          ]\n",
      "  C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(408): note: while compiling class template member function 'auto stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>::operator ()(const Eigen::Matrix<stan::math::var,-1,1,0,-1,1> &)'\n",
      "          with\n",
      "          [\n",
      "              F=stan::math::internal::simplex_constrain_op\n",
      "          ]\n",
      "  C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(617): note: see reference to class template instantiation 'stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>' being compiled\n",
      "          with\n",
      "          [\n",
      "              F=stan::math::internal::simplex_constrain_op\n",
      "          ]\n",
      "  C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/fun/simplex_constrain.hpp(97): note: see reference to function template instantiation 'auto stan::math::adj_jac_apply<stan::math::internal::simplex_constrain_op,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>(const Eigen::Matrix<stan::math::var,-1,1,0,-1,1> &)' being compiled\n",
      "  C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(412): error C2660: 'stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>::prepare_x_vis': function does not take 1 arguments\n",
      "          with\n",
      "          [\n",
      "              F=stan::math::internal::simplex_constrain_op\n",
      "          ]\n",
      "  C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(389): note: see declaration of 'stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>::prepare_x_vis'\n",
      "          with\n",
      "          [\n",
      "              F=stan::math::internal::simplex_constrain_op\n",
      "          ]\n",
      "  C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(414): error C2664: 'stan::math::var stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>::build_return_varis_and_vars(const double &)': cannot convert argument 1 from 'Eigen::VectorXd' to 'const double &'\n",
      "          with\n",
      "          [\n",
      "              F=stan::math::internal::simplex_constrain_op\n",
      "          ]\n",
      "  C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(414): note: Reason: cannot convert from 'Eigen::VectorXd' to 'const double'\n",
      "  C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(414): note: No user-defined-conversion operator available that can perform this conversion, or the operator cannot be called\n",
      "  C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(333): note: see declaration of 'stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>::build_return_varis_and_vars'\n",
      "          with\n",
      "          [\n",
      "              F=stan::math::internal::simplex_constrain_op\n",
      "          ]\n",
      "  C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/fun/simplex_constrain.hpp(97): error C2440: 'return': cannot convert from 'void' to 'Eigen::Matrix<stan::math::var,-1,1,0,-1,1>'\n",
      "  C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/fun/simplex_constrain.hpp(97): note: Expressions of type void cannot be converted to other types\n",
      "  C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(410): error C2660: 'stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>::count_memory': function does not take 2 arguments\n",
      "          with\n",
      "          [\n",
      "              F=stan::math::internal::softmax_op\n",
      "          ]\n",
      "  C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(246): note: see declaration of 'stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>::count_memory'\n",
      "          with\n",
      "          [\n",
      "              F=stan::math::internal::softmax_op\n",
      "          ]\n",
      "  C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(408): note: while compiling class template member function 'auto stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>::operator ()(const Eigen::Matrix<stan::math::var,-1,1,0,-1,1> &)'\n",
      "          with\n",
      "          [\n",
      "              F=stan::math::internal::softmax_op\n",
      "          ]\n",
      "  C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(617): note: see reference to class template instantiation 'stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>' being compiled\n",
      "          with\n",
      "          [\n",
      "              F=stan::math::internal::softmax_op\n",
      "          ]\n",
      "  C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/fun/softmax.hpp(78): note: see reference to function template instantiation 'auto stan::math::adj_jac_apply<stan::math::internal::softmax_op,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>(const Eigen::Matrix<stan::math::var,-1,1,0,-1,1> &)' being compiled\n",
      "  C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(412): error C2660: 'stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>::prepare_x_vis': function does not take 1 arguments\n",
      "          with\n",
      "          [\n",
      "              F=stan::math::internal::softmax_op\n",
      "          ]\n",
      "  C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(389): note: see declaration of 'stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>::prepare_x_vis'\n",
      "          with\n",
      "          [\n",
      "              F=stan::math::internal::softmax_op\n",
      "          ]\n",
      "  C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(414): error C2664: 'stan::math::var stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>::build_return_varis_and_vars(const double &)': cannot convert argument 1 from 'Eigen::VectorXd' to 'const double &'\n",
      "          with\n",
      "          [\n",
      "              F=stan::math::internal::softmax_op\n",
      "          ]\n",
      "  C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(414): note: Reason: cannot convert from 'Eigen::VectorXd' to 'const double'\n",
      "  C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(414): note: No user-defined-conversion operator available that can perform this conversion, or the operator cannot be called\n",
      "  C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(333): note: see declaration of 'stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>::build_return_varis_and_vars'\n",
      "          with\n",
      "          [\n",
      "              F=stan::math::internal::softmax_op\n",
      "          ]\n",
      "  C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/fun/softmax.hpp(78): error C2440: 'return': cannot convert from 'void' to 'Eigen::Matrix<stan::math::var,-1,1,0,-1,1>'\n",
      "  C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/fun/softmax.hpp(78): note: Expressions of type void cannot be converted to other types\n",
      "  C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\src\\stan/mcmc/sample.hpp(33): warning C4244: 'return': conversion from 'Eigen::EigenBase<Derived>::Index' to 'int', possible loss of data\n",
      "          with\n",
      "          [\n",
      "              Derived=Eigen::Matrix<double,-1,1,0,-1,1>\n",
      "          ]\n",
      "  C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\src\\stan/services/util/read_dense_inv_metric.hpp(38): warning C4267: 'argument': conversion from 'size_t' to 'int', possible loss of data\n",
      "  C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\src\\stan/mcmc/windowed_adaptation.hpp(49): warning C4244: '=': conversion from 'double' to 'unsigned int', possible loss of data\n",
      "  C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\src\\stan/mcmc/windowed_adaptation.hpp(50): warning C4244: '=': conversion from 'double' to 'unsigned int', possible loss of data\n",
      "  C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\src\\stan/variational/print_progress.hpp(52): warning C4244: 'initializing': conversion from 'double' to 'int', possible loss of data\n",
      "  C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\src\\stan/variational/families/normal_fullrank.hpp(91): warning C4267: 'initializing': conversion from 'size_t' to 'const int', possible loss of data\n",
      "  C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\src\\stan/variational/families/normal_fullrank.hpp(105): warning C4244: 'initializing': conversion from 'Eigen::EigenBase<Derived>::Index' to 'const int', possible loss of data\n",
      "          with\n",
      "          [\n",
      "              Derived=Eigen::Matrix<double,-1,1,0,-1,1>\n",
      "          ]\n",
      "  C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\src\\stan/variational/families/normal_fullrank.hpp(124): warning C4244: 'initializing': conversion from 'Eigen::EigenBase<Derived>::Index' to 'const int', possible loss of data\n",
      "          with\n",
      "          [\n",
      "              Derived=Eigen::Matrix<double,-1,1,0,-1,1>\n",
      "          ]\n",
      "  C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\src\\stan/variational/families/normal_meanfield.hpp(48): warning C4267: 'initializing': conversion from 'size_t' to 'const int', possible loss of data\n",
      "  C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\src\\stan/variational/families/normal_meanfield.hpp(61): warning C4244: 'initializing': conversion from 'Eigen::EigenBase<Derived>::Index' to 'const int', possible loss of data\n",
      "          with\n",
      "          [\n",
      "              Derived=Eigen::Matrix<double,-1,1,0,-1,1>\n",
      "          ]\n",
      "  C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\src\\stan/variational/families/normal_meanfield.hpp(76): warning C4244: 'initializing': conversion from 'Eigen::EigenBase<Derived>::Index' to 'const int', possible loss of data\n",
      "          with\n",
      "          [\n",
      "              Derived=Eigen::Matrix<double,-1,1,0,-1,1>\n",
      "          ]\n",
      "  C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\src\\stan/io/cmd_line.hpp(32): warning C4267: 'initializing': conversion from 'size_t' to 'int', possible loss of data\n",
      "  C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\src\\stan/lang/rethrow_located.hpp(117): warning C4267: 'initializing': conversion from 'size_t' to 'int', possible loss of data\n",
      "  C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\src\\stan/model/indexing/rvalue_index_size.hpp(20): warning C4267: 'return': conversion from 'size_t' to 'int', possible loss of data\n",
      "  C:\\Users\\vtiwari2\\AppData\\Local\\Temp\\1\\pystan_gcqhcxp9\\stanfit4anon_model_f5236004a3fd5b8429270d00efcc0cf9_5748532310016087196.cpp(5077): warning C4267: '=': conversion from 'size_t' to 'int', possible loss of data\n",
      "  error: command 'C:\\\\Program Files (x86)\\\\Microsoft Visual Studio\\\\2019\\\\BuildTools\\\\VC\\\\Tools\\\\MSVC\\\\14.29.30133\\\\bin\\\\HostX86\\\\x64\\\\cl.exe' failed with exit status 2\n",
      "  ----------------------------------------\n",
      "  ERROR: Failed building wheel for fbprophet\n",
      "    ERROR: Command errored out with exit status 1:\n",
      "     command: 'C:\\Users\\vtiwari2\\Anaconda3\\python.exe' -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\vtiwari2\\\\AppData\\\\Local\\\\Temp\\\\1\\\\pip-install-7ei2nuta\\\\fbprophet_1b01894b54ea43c0bb0225d22ec02073\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\vtiwari2\\\\AppData\\\\Local\\\\Temp\\\\1\\\\pip-install-7ei2nuta\\\\fbprophet_1b01894b54ea43c0bb0225d22ec02073\\\\setup.py'\"'\"';f=getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__);code=f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' install --record 'C:\\Users\\vtiwari2\\AppData\\Local\\Temp\\1\\pip-record-4gk4v2sc\\install-record.txt' --single-version-externally-managed --compile --install-headers 'C:\\Users\\vtiwari2\\Anaconda3\\Include\\fbprophet'\n",
      "         cwd: C:\\Users\\vtiwari2\\AppData\\Local\\Temp\\1\\pip-install-7ei2nuta\\fbprophet_1b01894b54ea43c0bb0225d22ec02073\\\n",
      "    Complete output (285 lines):\n",
      "    running install\n",
      "    running build\n",
      "    running build_py\n",
      "    creating build\n",
      "    creating build\\lib\n",
      "    creating build\\lib\\fbprophet\n",
      "    creating build\\lib\\fbprophet\\stan_model\n",
      "    Importing plotly failed. Interactive plots will not work.\n",
      "    INFO:pystan:COMPILING THE C++ CODE FOR MODEL anon_model_f5236004a3fd5b8429270d00efcc0cf9 NOW.\n",
      "    WARNING:pystan:MSVC compiler is not supported\n",
      "    stanfit4anon_model_f5236004a3fd5b8429270d00efcc0cf9_6243042721993354468.cpp\n",
      "    C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\numpy\\core\\include\\numpy\\npy_1_7_deprecated_api.h(14) : Warning Msg: Using deprecated NumPy API, disable it with #define NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION\n",
      "    C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/prim/mat/meta/seq_view.hpp(145): warning C4267: 'return': conversion from 'size_t' to 'int', possible loss of data\n",
      "    C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/prim/mat/fun/multiply_lower_tri_self_transpose.hpp(19): warning C4244: 'initializing': conversion from 'Eigen::EigenBase<Derived>::Index' to 'int', possible loss of data\n",
      "            with\n",
      "            [\n",
      "                Derived=Eigen::Matrix<double,-1,-1,0,-1,-1>\n",
      "            ]\n",
      "    C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/prim/mat/fun/multiply_lower_tri_self_transpose.hpp(27): warning C4244: 'initializing': conversion from 'Eigen::EigenBase<Derived>::Index' to 'int', possible loss of data\n",
      "            with\n",
      "            [\n",
      "                Derived=Eigen::Matrix<double,-1,-1,0,-1,-1>\n",
      "            ]\n",
      "    C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/prim/mat/fun/csr_u_to_z.hpp(28): warning C4267: 'argument': conversion from 'size_t' to 'int', possible loss of data\n",
      "    C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/prim/mat/fun/matrix_exp_action_handler.hpp(118): warning C4244: 'initializing': conversion from 'double' to 'int', possible loss of data\n",
      "    C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/prim/mat/fun/to_matrix.hpp(136): warning C4267: 'initializing': conversion from 'size_t' to 'int', possible loss of data\n",
      "    C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/prim/mat/fun/to_row_vector.hpp(31): warning C4267: 'initializing': conversion from 'size_t' to 'int', possible loss of data\n",
      "    C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/prim/mat/fun/to_vector.hpp(31): warning C4267: 'initializing': conversion from 'size_t' to 'int', possible loss of data\n",
      "    C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/prim/mat/fun/welford_covar_estimator.hpp(31): warning C4244: 'return': conversion from 'double' to 'int', possible loss of data\n",
      "    C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/prim/mat/fun/welford_var_estimator.hpp(31): warning C4244: 'return': conversion from 'double' to 'int', possible loss of data\n",
      "    C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\src\\stan/io/dump.hpp(243): warning C4101: 'exc': unreferenced local variable\n",
      "    C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\src\\stan/io/dump.hpp(270): warning C4101: 'exc': unreferenced local variable\n",
      "    C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\src\\stan/io/dump.hpp(284): warning C4101: 'exc': unreferenced local variable\n",
      "    C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/meta/operands_and_partials.hpp(42): warning C4267: 'return': conversion from 'size_t' to 'int', possible loss of data\n",
      "    C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/meta/operands_and_partials.hpp(155): warning C4267: 'return': conversion from 'size_t' to 'int', possible loss of data\n",
      "    C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/scal/fun/bessel_first_kind.hpp(19): warning C4244: 'argument': conversion from 'double' to 'int', possible loss of data\n",
      "    C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/scal/fun/bessel_first_kind.hpp(20): warning C4244: 'argument': conversion from 'double' to 'int', possible loss of data\n",
      "    C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/scal/fun/bessel_second_kind.hpp(19): warning C4244: 'argument': conversion from 'double' to 'int', possible loss of data\n",
      "    C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/scal/fun/bessel_second_kind.hpp(20): warning C4244: 'argument': conversion from 'double' to 'int', possible loss of data\n",
      "    C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/scal/fun/modified_bessel_first_kind.hpp(19): warning C4244: 'argument': conversion from 'double' to 'int', possible loss of data\n",
      "    C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/scal/fun/modified_bessel_first_kind.hpp(20): warning C4244: 'argument': conversion from 'double' to 'int', possible loss of data\n",
      "    C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/scal/fun/modified_bessel_second_kind.hpp(19): warning C4244: 'argument': conversion from 'double' to 'int', possible loss of data\n",
      "    C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/scal/fun/modified_bessel_second_kind.hpp(20): warning C4244: 'argument': conversion from 'double' to 'int', possible loss of data\n",
      "    C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/scal/fun/rising_factorial.hpp(18): warning C4244: 'argument': conversion from 'double' to 'int', possible loss of data\n",
      "    C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/fun/cholesky_decompose.hpp(82): warning C4244: 'initializing': conversion from 'Eigen::EigenBase<Derived>::Index' to 'int', possible loss of data\n",
      "            with\n",
      "            [\n",
      "                Derived=Eigen::Matrix<stan::math::var,-1,-1,0,-1,-1>\n",
      "            ]\n",
      "    C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/fun/cholesky_decompose.hpp(194): warning C4244: 'initializing': conversion from 'Eigen::EigenBase<Derived>::Index' to 'int', possible loss of data\n",
      "            with\n",
      "            [\n",
      "                Derived=Eigen::Matrix<stan::math::var,-1,-1,0,-1,-1>\n",
      "            ]\n",
      "    C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/fun/log_softmax.hpp(104): warning C4244: 'argument': conversion from 'Eigen::EigenBase<Derived>::Index' to 'int', possible loss of data\n",
      "            with\n",
      "            [\n",
      "                Derived=Eigen::Matrix<stan::math::var,-1,1,0,-1,1>\n",
      "            ]\n",
      "    C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/fun/multiply_lower_tri_self_transpose.hpp(21): warning C4244: 'initializing': conversion from 'Eigen::EigenBase<Derived>::Index' to 'int', possible loss of data\n",
      "            with\n",
      "            [\n",
      "                Derived=Eigen::Matrix<stan::math::var,-1,-1,0,-1,-1>\n",
      "            ]\n",
      "    C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/fun/multiply_lower_tri_self_transpose.hpp(22): warning C4244: 'initializing': conversion from 'Eigen::EigenBase<Derived>::Index' to 'int', possible loss of data\n",
      "            with\n",
      "            [\n",
      "                Derived=Eigen::Matrix<stan::math::var,-1,-1,0,-1,-1>\n",
      "            ]\n",
      "    C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(410): error C2660: 'stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>::count_memory': function does not take 2 arguments\n",
      "            with\n",
      "            [\n",
      "                F=stan::math::internal::ordered_constrain_op\n",
      "            ]\n",
      "    C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(246): note: see declaration of 'stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>::count_memory'\n",
      "            with\n",
      "            [\n",
      "                F=stan::math::internal::ordered_constrain_op\n",
      "            ]\n",
      "    C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(408): note: while compiling class template member function 'auto stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>::operator ()(const Eigen::Matrix<stan::math::var,-1,1,0,-1,1> &)'\n",
      "            with\n",
      "            [\n",
      "                F=stan::math::internal::ordered_constrain_op\n",
      "            ]\n",
      "    C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(617): note: see reference to class template instantiation 'stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>' being compiled\n",
      "            with\n",
      "            [\n",
      "                F=stan::math::internal::ordered_constrain_op\n",
      "            ]\n",
      "    C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/fun/ordered_constrain.hpp(88): note: see reference to function template instantiation 'auto stan::math::adj_jac_apply<stan::math::internal::ordered_constrain_op,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>(const Eigen::Matrix<stan::math::var,-1,1,0,-1,1> &)' being compiled\n",
      "    C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(412): error C2660: 'stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>::prepare_x_vis': function does not take 1 arguments\n",
      "            with\n",
      "            [\n",
      "                F=stan::math::internal::ordered_constrain_op\n",
      "            ]\n",
      "    C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(389): note: see declaration of 'stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>::prepare_x_vis'\n",
      "            with\n",
      "            [\n",
      "                F=stan::math::internal::ordered_constrain_op\n",
      "            ]\n",
      "    C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(414): error C2664: 'stan::math::var stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>::build_return_varis_and_vars(const double &)': cannot convert argument 1 from 'Eigen::VectorXd' to 'const double &'\n",
      "            with\n",
      "            [\n",
      "                F=stan::math::internal::ordered_constrain_op\n",
      "            ]\n",
      "    C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(414): note: Reason: cannot convert from 'Eigen::VectorXd' to 'const double'\n",
      "    C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(414): note: No user-defined-conversion operator available that can perform this conversion, or the operator cannot be called\n",
      "    C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(333): note: see declaration of 'stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>::build_return_varis_and_vars'\n",
      "            with\n",
      "            [\n",
      "                F=stan::math::internal::ordered_constrain_op\n",
      "            ]\n",
      "    C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/fun/ordered_constrain.hpp(88): error C2440: 'return': cannot convert from 'void' to 'Eigen::Matrix<stan::math::var,-1,1,0,-1,1>'\n",
      "    C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/fun/ordered_constrain.hpp(88): note: Expressions of type void cannot be converted to other types\n",
      "    C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(410): error C2660: 'stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>::count_memory': function does not take 2 arguments\n",
      "            with\n",
      "            [\n",
      "                F=stan::math::internal::positive_ordered_constrain_op\n",
      "            ]\n",
      "    C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(246): note: see declaration of 'stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>::count_memory'\n",
      "            with\n",
      "            [\n",
      "                F=stan::math::internal::positive_ordered_constrain_op\n",
      "            ]\n",
      "    C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(408): note: while compiling class template member function 'auto stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>::operator ()(const Eigen::Matrix<stan::math::var,-1,1,0,-1,1> &)'\n",
      "            with\n",
      "            [\n",
      "                F=stan::math::internal::positive_ordered_constrain_op\n",
      "            ]\n",
      "    C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(617): note: see reference to class template instantiation 'stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>' being compiled\n",
      "            with\n",
      "            [\n",
      "                F=stan::math::internal::positive_ordered_constrain_op\n",
      "            ]\n",
      "    C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/fun/positive_ordered_constrain.hpp(86): note: see reference to function template instantiation 'auto stan::math::adj_jac_apply<stan::math::internal::positive_ordered_constrain_op,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>(const Eigen::Matrix<stan::math::var,-1,1,0,-1,1> &)' being compiled\n",
      "    C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(412): error C2660: 'stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>::prepare_x_vis': function does not take 1 arguments\n",
      "            with\n",
      "            [\n",
      "                F=stan::math::internal::positive_ordered_constrain_op\n",
      "            ]\n",
      "    C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(389): note: see declaration of 'stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>::prepare_x_vis'\n",
      "            with\n",
      "            [\n",
      "                F=stan::math::internal::positive_ordered_constrain_op\n",
      "            ]\n",
      "    C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(414): error C2664: 'stan::math::var stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>::build_return_varis_and_vars(const double &)': cannot convert argument 1 from 'Eigen::VectorXd' to 'const double &'\n",
      "            with\n",
      "            [\n",
      "                F=stan::math::internal::positive_ordered_constrain_op\n",
      "            ]\n",
      "    C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(414): note: Reason: cannot convert from 'Eigen::VectorXd' to 'const double'\n",
      "    C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(414): note: No user-defined-conversion operator available that can perform this conversion, or the operator cannot be called\n",
      "    C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(333): note: see declaration of 'stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>::build_return_varis_and_vars'\n",
      "            with\n",
      "            [\n",
      "                F=stan::math::internal::positive_ordered_constrain_op\n",
      "            ]\n",
      "    C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/fun/positive_ordered_constrain.hpp(86): error C2440: 'return': cannot convert from 'void' to 'Eigen::Matrix<stan::math::var,-1,1,0,-1,1>'\n",
      "    C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/fun/positive_ordered_constrain.hpp(86): note: Expressions of type void cannot be converted to other types\n",
      "    C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(410): error C2660: 'stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>::count_memory': function does not take 2 arguments\n",
      "            with\n",
      "            [\n",
      "                F=stan::math::internal::simplex_constrain_op\n",
      "            ]\n",
      "    C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(246): note: see declaration of 'stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>::count_memory'\n",
      "            with\n",
      "            [\n",
      "                F=stan::math::internal::simplex_constrain_op\n",
      "            ]\n",
      "    C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(408): note: while compiling class template member function 'auto stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>::operator ()(const Eigen::Matrix<stan::math::var,-1,1,0,-1,1> &)'\n",
      "            with\n",
      "            [\n",
      "                F=stan::math::internal::simplex_constrain_op\n",
      "            ]\n",
      "    C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(617): note: see reference to class template instantiation 'stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>' being compiled\n",
      "            with\n",
      "            [\n",
      "                F=stan::math::internal::simplex_constrain_op\n",
      "            ]\n",
      "    C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/fun/simplex_constrain.hpp(97): note: see reference to function template instantiation 'auto stan::math::adj_jac_apply<stan::math::internal::simplex_constrain_op,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>(const Eigen::Matrix<stan::math::var,-1,1,0,-1,1> &)' being compiled\n",
      "    C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(412): error C2660: 'stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>::prepare_x_vis': function does not take 1 arguments\n",
      "            with\n",
      "            [\n",
      "                F=stan::math::internal::simplex_constrain_op\n",
      "            ]\n",
      "    C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(389): note: see declaration of 'stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>::prepare_x_vis'\n",
      "            with\n",
      "            [\n",
      "                F=stan::math::internal::simplex_constrain_op\n",
      "            ]\n",
      "    C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(414): error C2664: 'stan::math::var stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>::build_return_varis_and_vars(const double &)': cannot convert argument 1 from 'Eigen::VectorXd' to 'const double &'\n",
      "            with\n",
      "            [\n",
      "                F=stan::math::internal::simplex_constrain_op\n",
      "            ]\n",
      "    C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(414): note: Reason: cannot convert from 'Eigen::VectorXd' to 'const double'\n",
      "    C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(414): note: No user-defined-conversion operator available that can perform this conversion, or the operator cannot be called\n",
      "    C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(333): note: see declaration of 'stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>::build_return_varis_and_vars'\n",
      "            with\n",
      "            [\n",
      "                F=stan::math::internal::simplex_constrain_op\n",
      "            ]\n",
      "    C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/fun/simplex_constrain.hpp(97): error C2440: 'return': cannot convert from 'void' to 'Eigen::Matrix<stan::math::var,-1,1,0,-1,1>'\n",
      "    C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/fun/simplex_constrain.hpp(97): note: Expressions of type void cannot be converted to other types\n",
      "    C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(410): error C2660: 'stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>::count_memory': function does not take 2 arguments\n",
      "            with\n",
      "            [\n",
      "                F=stan::math::internal::softmax_op\n",
      "            ]\n",
      "    C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(246): note: see declaration of 'stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>::count_memory'\n",
      "            with\n",
      "            [\n",
      "                F=stan::math::internal::softmax_op\n",
      "            ]\n",
      "    C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(408): note: while compiling class template member function 'auto stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>::operator ()(const Eigen::Matrix<stan::math::var,-1,1,0,-1,1> &)'\n",
      "            with\n",
      "            [\n",
      "                F=stan::math::internal::softmax_op\n",
      "            ]\n",
      "    C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(617): note: see reference to class template instantiation 'stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>' being compiled\n",
      "            with\n",
      "            [\n",
      "                F=stan::math::internal::softmax_op\n",
      "            ]\n",
      "    C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/fun/softmax.hpp(78): note: see reference to function template instantiation 'auto stan::math::adj_jac_apply<stan::math::internal::softmax_op,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>(const Eigen::Matrix<stan::math::var,-1,1,0,-1,1> &)' being compiled\n",
      "    C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(412): error C2660: 'stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>::prepare_x_vis': function does not take 1 arguments\n",
      "            with\n",
      "            [\n",
      "                F=stan::math::internal::softmax_op\n",
      "            ]\n",
      "    C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(389): note: see declaration of 'stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>::prepare_x_vis'\n",
      "            with\n",
      "            [\n",
      "                F=stan::math::internal::softmax_op\n",
      "            ]\n",
      "    C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(414): error C2664: 'stan::math::var stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>::build_return_varis_and_vars(const double &)': cannot convert argument 1 from 'Eigen::VectorXd' to 'const double &'\n",
      "            with\n",
      "            [\n",
      "                F=stan::math::internal::softmax_op\n",
      "            ]\n",
      "    C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(414): note: Reason: cannot convert from 'Eigen::VectorXd' to 'const double'\n",
      "    C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(414): note: No user-defined-conversion operator available that can perform this conversion, or the operator cannot be called\n",
      "    C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(333): note: see declaration of 'stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>::build_return_varis_and_vars'\n",
      "            with\n",
      "            [\n",
      "                F=stan::math::internal::softmax_op\n",
      "            ]\n",
      "    C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/fun/softmax.hpp(78): error C2440: 'return': cannot convert from 'void' to 'Eigen::Matrix<stan::math::var,-1,1,0,-1,1>'\n",
      "    C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/fun/softmax.hpp(78): note: Expressions of type void cannot be converted to other types\n",
      "    C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\src\\stan/mcmc/sample.hpp(33): warning C4244: 'return': conversion from 'Eigen::EigenBase<Derived>::Index' to 'int', possible loss of data\n",
      "            with\n",
      "            [\n",
      "                Derived=Eigen::Matrix<double,-1,1,0,-1,1>\n",
      "            ]\n",
      "    C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\src\\stan/services/util/read_dense_inv_metric.hpp(38): warning C4267: 'argument': conversion from 'size_t' to 'int', possible loss of data\n",
      "    C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\src\\stan/mcmc/windowed_adaptation.hpp(49): warning C4244: '=': conversion from 'double' to 'unsigned int', possible loss of data\n",
      "    C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\src\\stan/mcmc/windowed_adaptation.hpp(50): warning C4244: '=': conversion from 'double' to 'unsigned int', possible loss of data\n",
      "    C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\src\\stan/variational/print_progress.hpp(52): warning C4244: 'initializing': conversion from 'double' to 'int', possible loss of data\n",
      "    C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\src\\stan/variational/families/normal_fullrank.hpp(91): warning C4267: 'initializing': conversion from 'size_t' to 'const int', possible loss of data\n",
      "    C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\src\\stan/variational/families/normal_fullrank.hpp(105): warning C4244: 'initializing': conversion from 'Eigen::EigenBase<Derived>::Index' to 'const int', possible loss of data\n",
      "            with\n",
      "            [\n",
      "                Derived=Eigen::Matrix<double,-1,1,0,-1,1>\n",
      "            ]\n",
      "    C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\src\\stan/variational/families/normal_fullrank.hpp(124): warning C4244: 'initializing': conversion from 'Eigen::EigenBase<Derived>::Index' to 'const int', possible loss of data\n",
      "            with\n",
      "            [\n",
      "                Derived=Eigen::Matrix<double,-1,1,0,-1,1>\n",
      "            ]\n",
      "    C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\src\\stan/variational/families/normal_meanfield.hpp(48): warning C4267: 'initializing': conversion from 'size_t' to 'const int', possible loss of data\n",
      "    C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\src\\stan/variational/families/normal_meanfield.hpp(61): warning C4244: 'initializing': conversion from 'Eigen::EigenBase<Derived>::Index' to 'const int', possible loss of data\n",
      "            with\n",
      "            [\n",
      "                Derived=Eigen::Matrix<double,-1,1,0,-1,1>\n",
      "            ]\n",
      "    C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\src\\stan/variational/families/normal_meanfield.hpp(76): warning C4244: 'initializing': conversion from 'Eigen::EigenBase<Derived>::Index' to 'const int', possible loss of data\n",
      "            with\n",
      "            [\n",
      "                Derived=Eigen::Matrix<double,-1,1,0,-1,1>\n",
      "            ]\n",
      "    C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\src\\stan/io/cmd_line.hpp(32): warning C4267: 'initializing': conversion from 'size_t' to 'int', possible loss of data\n",
      "    C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\src\\stan/lang/rethrow_located.hpp(117): warning C4267: 'initializing': conversion from 'size_t' to 'int', possible loss of data\n",
      "    C:\\Users\\vtiwari2\\Anaconda3\\lib\\site-packages\\pystan\\stan\\src\\stan/model/indexing/rvalue_index_size.hpp(20): warning C4267: 'return': conversion from 'size_t' to 'int', possible loss of data\n",
      "    C:\\Users\\vtiwari2\\AppData\\Local\\Temp\\1\\pystan_ig74gzo6\\stanfit4anon_model_f5236004a3fd5b8429270d00efcc0cf9_6243042721993354468.cpp(5077): warning C4267: '=': conversion from 'size_t' to 'int', possible loss of data\n",
      "    error: command 'C:\\\\Program Files (x86)\\\\Microsoft Visual Studio\\\\2019\\\\BuildTools\\\\VC\\\\Tools\\\\MSVC\\\\14.29.30133\\\\bin\\\\HostX86\\\\x64\\\\cl.exe' failed with exit status 2\n",
      "    ----------------------------------------\n",
      "ERROR: Command errored out with exit status 1: 'C:\\Users\\vtiwari2\\Anaconda3\\python.exe' -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\vtiwari2\\\\AppData\\\\Local\\\\Temp\\\\1\\\\pip-install-7ei2nuta\\\\fbprophet_1b01894b54ea43c0bb0225d22ec02073\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\vtiwari2\\\\AppData\\\\Local\\\\Temp\\\\1\\\\pip-install-7ei2nuta\\\\fbprophet_1b01894b54ea43c0bb0225d22ec02073\\\\setup.py'\"'\"';f=getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__);code=f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' install --record 'C:\\Users\\vtiwari2\\AppData\\Local\\Temp\\1\\pip-record-4gk4v2sc\\install-record.txt' --single-version-externally-managed --compile --install-headers 'C:\\Users\\vtiwari2\\Anaconda3\\Include\\fbprophet' Check the logs for full command output.\n"
     ]
    }
   ],
   "source": [
    "!pip install pystan\n",
    "!pip install fbprophet"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "ename": "ModuleNotFoundError",
     "evalue": "No module named 'fbprophet'",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mModuleNotFoundError\u001b[0m                       Traceback (most recent call last)",
      "\u001b[1;32m<ipython-input-6-7bda92cab60d>\u001b[0m in \u001b[0;36m<module>\u001b[1;34m\u001b[0m\n\u001b[0;32m      1\u001b[0m \u001b[1;32mimport\u001b[0m \u001b[0mpandas\u001b[0m \u001b[1;32mas\u001b[0m \u001b[0mpd\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m----> 2\u001b[1;33m \u001b[1;32mfrom\u001b[0m \u001b[0mfbprophet\u001b[0m \u001b[1;32mimport\u001b[0m \u001b[0mProphet\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[1;31mModuleNotFoundError\u001b[0m: No module named 'fbprophet'"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "from fbprophet import Prophet"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 1. Read in Data and Process Dates"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Time Date</th>\n",
       "      <th>Product</th>\n",
       "      <th>Store</th>\n",
       "      <th>Value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1012018</td>\n",
       "      <td>2667437</td>\n",
       "      <td>QLD_CW_ST0203</td>\n",
       "      <td>2926.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2012018</td>\n",
       "      <td>2667437</td>\n",
       "      <td>QLD_CW_ST0203</td>\n",
       "      <td>2687.531</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3012018</td>\n",
       "      <td>2667437</td>\n",
       "      <td>QLD_CW_ST0203</td>\n",
       "      <td>2793.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4012018</td>\n",
       "      <td>2667437</td>\n",
       "      <td>QLD_CW_ST0203</td>\n",
       "      <td>2394.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5012018</td>\n",
       "      <td>2667437</td>\n",
       "      <td>QLD_CW_ST0203</td>\n",
       "      <td>2660.000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Time Date  Product          Store     Value\n",
       "0    1012018  2667437  QLD_CW_ST0203  2926.000\n",
       "1    2012018  2667437  QLD_CW_ST0203  2687.531\n",
       "2    3012018  2667437  QLD_CW_ST0203  2793.000\n",
       "3    4012018  2667437  QLD_CW_ST0203  2394.000\n",
       "4    5012018  2667437  QLD_CW_ST0203  2660.000"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.read_csv('dataset.csv')\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "df['Year'] = df['Time Date'].apply(lambda x: str(x)[-4:])\n",
    "df['Month'] = df['Time Date'].apply(lambda x: str(x)[-6:-4])\n",
    "df['Day'] = df['Time Date'].apply(lambda x: str(x)[:-6])\n",
    "df['ds'] = pd.DatetimeIndex(df['Year']+'-'+df['Month']+'-'+df['Day'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>y</th>\n",
       "      <th>ds</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2926.000</td>\n",
       "      <td>2018-01-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2687.531</td>\n",
       "      <td>2018-01-02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2793.000</td>\n",
       "      <td>2018-01-03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>2394.000</td>\n",
       "      <td>2018-01-04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>2660.000</td>\n",
       "      <td>2018-01-05</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          y         ds\n",
       "0  2926.000 2018-01-01\n",
       "1  2687.531 2018-01-02\n",
       "2  2793.000 2018-01-03\n",
       "3  2394.000 2018-01-04\n",
       "4  2660.000 2018-01-05"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = df.loc[(df['Product']==2667437) & (df['Store']=='QLD_CW_ST0203')]\n",
    "df.drop(['Time Date', 'Product', 'Store', 'Year', 'Month', 'Day'], axis=1, inplace=True)\n",
    "df.columns = ['y', 'ds']\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 2. Train Model"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "m = Prophet(interval_width=0.95, daily_seasonality=True)\n",
    "model = m.fit(df)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 3. Forecast Away"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "future = m.make_future_dataframe(periods=100,freq='D')\n",
    "forecast = m.predict(future)\n",
    "forecast.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "plot1 = m.plot(forecast)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "plt2 = m.plot_components(forecast)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
